xdsl/frontend/code_generation.py

```diff
@@ -71,6 +72,87 @@ def visit_Assign(self, node: ast.Assign) -> None:
         # TODO: Implement assignemnt in the next patch.
         pass
 
+    def visit_BinOp(self, node: ast.BinOp):
+        op_name: str = node.op.__class__.__name__
+
+        # Local table which maps the name of a binry operator to the
+        # corresponding Python function.
+        op_to_python = {
+            "Add": "__add__",
+            "BitAnd": "__and__",
+            "LShift": "__lshift__",
+            "Mult": "__mul__",
+            "RShift": "__rshift__",
+            "Sub": "__sub__",
+        }
+        if op_name not in op_to_python:
+            raise CodeGenerationException(
+                node.lineno, node.col_offset,
+                f"Unknown binary operation {op_name}.")
+
+        self.visit(node.right)
+        rhs = self.inserter.get_operand()
+        self.visit(node.left)
+        lhs = self.inserter.get_operand()
+        if lhs.typ != rhs.typ:
+            raise CodeGenerationException(
+                node.lineno, node.col_offset,
+                f"Expected the same types for binary operation '{op_name}', "
+                f"but got {lhs.typ} and {rhs.typ}.")
+
+        # Look-up what is the frontend type we deal with to resolve the binary
+        # operation.
+        frontend_type = self.type_converter.xdsl_to_frontend_type_map[
+            lhs.typ.__class__]
+
+        op = OpResolver.resolve_op_overload(op_to_python[op_name],
+                                            frontend_type)(lhs, rhs)
+        self.inserter.insert_op(op)
+
+    def visit_Compare(self, node: ast.Compare):
+        # Allow a single comparison only.
+        if len(node.comparators) != 1 or len(node.ops) != 1:
+            raise CodeGenerationException(
+                node.lineno, node.col_offset,
+                "Expected a single comparator, but found "
+                f"{len(node.comparators)}.")
+        comp = node.comparators[0]
+        op_name: str = node.ops[0].__class__.__name__
+
+        # Local table which maps the name of a comparison operator to the
+        # corresponding Python functions and xDSL mnemonics.
+        op_to_python_and_mnemonic = {
+            "Eq": ("__eq__", "eq"),
+            "Gt": ("__gt__", "sgt"),
+            "GtE": ("__ge__", "sge"),
+            "Lt": ("__lt__", "slt"),
+            "LtE": ("__le__", "sle"),
+            "NotEq": ("__ne__", "ne"),
+        }
```

Comment:
Same as binary op list: rename to remove confusion & make complete so that error is more useful